File: test/cpp/test_read_channels.cpp
// Your First C++ Program

#include <iostream>
#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_read_channels" << std::endl;

    std::string filename = "";
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }

    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    std::cout << "Reading montages..." << std::endl;
    std::vector<CPSGFile::MONTAGE> montages = reader.getMontages();

    if( montages.size()==0 ) {
        std::cout << "ERROR No montage found" << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS Montage count:" << montages.size() << std::endl;
    }

    for(const auto& m: montages) {
        std::cout     << " Montages: " << m.MontageName << std::endl;
        
        std::vector<CPSGFile::CHANNEL> channels = reader.getChannels(m.Index);
        std::cout     << "  Channels (" << channels.size() << ")" << std::endl;
        
        for(const auto& ch: channels) {
            std::cout << "      " << ch.ChannelName << ":" << ch.TrueSampleFrequency <<"\n";
        }
    }

    std::cout << "DONE" << std::endl;
    return 0;
}

File: test/python/test_read_channels.py

import inspect
from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader

reader = HarmonieReader.HarmonieReader()
print("Test: test_read_channels")

filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

if (not res):
    print(f"ERROR Failed to open the file:{filename}")
    quit()
else:
    print("SUCCESS")


print("Reading montages...")
montages = reader.get_montages()

if( len(montages)==0 ):
    print("ERROR No montage found")
    quit()
else:
    print(f"SUCCESS Montage count:{len(montages)}")

for m in montages:
    print(f" Montage:{m.name}")
    
    channels = reader.get_channels(m.index)
    print(f"  Channels ({len(channels)}")
    
    for ch in channels:
        print(f"     {ch.name}:{ch.unit}")
